### PR TITLE
zkevm-circuits: move `mock` dependency to the `test` target

### PR DESCRIPTION
**Pull request by pinkiebell**

### CODE REVIEW

File: zkevm-circuits/Cargo.toml [hunk 1]
```diff
@@ -47,4 +47,4 @@ rand_chacha = "0.3"
 
 [features]
 default = []
-test = []
+test = ["dep:mock"]
```
Comment (Dhole) [inline zkevm-circuits/Cargo.toml:50]: I want to mention that I've never seen this, so if someone can provide documentation on the usage of `dep:` prefix, I would really appreciate it!

**Review by Dhole — APPROVED** [review]
LGTM!

**Review by ed255 — APPROVED** [review]
LGTM!

PS: @Dhole is my non-work account, I forgot I was logged with that account when I sent the review :smiling_face_with_tear:

File: zkevm-circuits/Cargo.toml (hunk 1 above)
Comment (pinkiebell) [reply zkevm-circuits/Cargo.toml]: Found no docs for this. I just removed the prefix :)

File: zkevm-circuits/Cargo.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ eth-types = { path = "../eth-types" }
 gadgets = { path = "../gadgets" }
 ethers-core = "0.17.0"
 ethers-signers = "0.17.0"
-mock = { path = "../mock" }
+mock = { path = "../mock", optional = true }
```
Comment (DreamWuGit) [inline zkevm-circuits/Cargo.toml:20]: anybody know `optional` functionality here or share some document for it? thanks .

Comment (ed255) [reply zkevm-circuits/Cargo.toml:20]: It is explained here, in the second example of the section: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies

**Review by ChihChengLiang — APPROVED** [review]
LGTM

**Pull request merged**